app/indexer/controllers.py

```diff
@@ -38,6 +38,9 @@ def from_file():
         # filename = secure_filename(file.filename)
         file.save(join(dir_path, "urls_to_index.txt"))
         return render_template('indexer/progress_file.html')
+    else:
+        flash(".txt file required","text")
```

Comment:
This is great. Just two small comments:
* the error message is not obvious to the user when it appears. How about making it red? See https://stackoverflow.com/questions/44569040/change-color-of-flask-flash-messages
* when the template reloads, the submit buttons end up going over the edge of the box at the bottom. You could just hack it by making the box a little bigger in the first place... Also, curiously, the number of recorded pages at the top of the page disappears. Not sure why.